### ISSUE TITLE
Same origin request yields server error

### ISSUE THREAD

**Issue by tobiasdiez** [NozomuIkuta/h3-cors#5]
Suppose you setup cors to allow cross-origin requests from nuxt:
```ts
defineCorsEventHandler({
    methods: ['GET', 'POST', 'OPTIONS'],
    allowHeaders: ['Content-Type'],
    origin: ['https://nuxtjs.org/'],
    credentials: true,
  })
```

If you then start a dev server and visit a route where these cors policies apply, you get
```
[nuxt] [request error] [unhandled] [500] [h3-cors] http://127.0.0.1:3000 is not allowed origin
  at ./node_modules/@nozomuikuta/h3-cors/dist/index.mjs:108:13  
  at ./.nuxt/dev/index.mjs:4098:5  
  at ./node_modules/h3/dist/index.mjs:518:14  
  at ./node_modules/h3/dist/index.mjs:518:14  
  at Object.handler (./node_modules/h3/dist/index.mjs:725:12)  
  at Object.handler (./node_modules/h3/dist/index.mjs:567:31)  
  at async Server.toNodeHandle (./node_modules/h3/dist/index.mjs:631:7)
```

This doesn't make any sense since it's a same-origin request.
This is triggered by this line here:
https://github.com/NozomuIkuta/h3-cors/blob/89da1d78c50f0159cd461d6ad393649d57a1c292/src/handler.ts#L12-L14

Instead of throwing a server-side error, the correct behavior would be to depend on the browsers implementation and to only modify the Access-Control-Allow-Origin header as follows (see https://github.com/expressjs/cors/blob/f038e7722838fd83935674aa8c5bf452766741fb/lib/index.js#L41-L68):
- If non specified in the options, return `*`
- If a single origin is specified in the options, return that origin (this is the case in the above example)
- If multiple origins are specified in the options, return the origin if its in the allowed list, otherwise false. Moreover, [append a Vary header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin#cors_and_caching).

**Comment by nozomuikuta** [comment]
@tobiasdiez 

Thank you for reporting the bug.

According to your report, all we need is to delete the if-condition and let the utility functions append headers correctly, don't we?

```typescript
export function defineCorsEventHandler (options: CorsOptions): ReturnType<typeof defineEventHandler> {
  const { preflight: { statusCode } } = resolveCorsOptions(options)

  return defineEventHandler((event) => {
    if (isPreflight(event)) {
      appendCorsPreflightHeaders(event, options)

      event.res.statusCode = statusCode
      event.res.setHeader('Content-Length', '0')
      event.res.end()
    } else {
      appendCorsActualRequestHeaders(event, options)
    }
  })
} 
```

**Comment by tobiasdiez** [comment]
Yes, and my reading of https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin#directives is that https://github.com/NozomuIkuta/h3-cors/blob/89da1d78c50f0159cd461d6ad393649d57a1c292/src/utils.ts#L62-L64 also needs to be changed to check if the request's origin is in the list of accepted origins (in CorsOptions), and only then add the particular Access-Control-Allow-Origin header.

**Comment by nozomuikuta** [comment]
@tobiasdiez 

I've just published v0.1.7 (diffs: https://github.com/NozomuIkuta/h3-cors/compare/v0.1.6...v0.1.7)

Sorry to bother you, but I do appreciate if you test the code below with your product. 🙏

**Comment by tobiasdiez** [comment]
Thanks for the quick fix! It works for me!
